File: src/Monad.AB.Domain.Entities/FormField.cs
using System;
using System.Collections.Generic;

namespace Monad.AB.Domain.Entities
{
    public class FormField : BaseEntity
    {
        public int FormID { get; set; }
        public int FieldTypeID { get; set; }
        public string Name { get; set; }
        public string Label { get; set; }
        public string Description { get; set; }
        public int Width { get; set; }
        public int RowNumber { get; set; }
        public int RowSpan { get; set; }
        public int ColumnNumber { get; set; }
        public int ColumnSpan { get; set; }
        public int GroupControlID { get; set; }
        public string GroupName { get; set; }
        public bool IsRequired { get; set; }
        public bool IsIdentifier { get; set; }
        public bool IsForeignKey { get; set; }
        public int? ForeignFieldID { get; set; }
        public DateTime CreatedDateUtc { get; set; }
        public DateTime LastModifiedDateUtc { get; set; }
        public int LastModifiedBy { get; set; }
        public  FieldType FieldType { get; set; }
        public IList<FormFieldView> Views { get; set; }
    }
}

File: src/Monad.AB.Web.ng/src/app/main/formFields/shared/formField.ts
﻿export class FormViewTypeModel {
    public id: number;
    public name: string;
    public description: string;
}
export class FormFieldView {
    public id: number;
    public formFieldId: number;
    public viewID: number;
    public fieldID: number;
    public row: number;
    public rowSpan: number;
    public column: number;
    public columnSpan: number;
    public readOnly: boolean;
    public hidden: boolean;
    public createdDateUtc: Date;
    public lastModifiedDateUtc: Date;
    public lastModifiedBy: number;
    public view: FormViewTypeModel;
}
export class FormField {
    public id: number;
    public formID: number;
    public fieldTypeID: number;
    public name: string;
    public label: string;
    public description: string;
    public width: number;
    public rowNumber: number;
    public rowSpan: number;
    public columnNumber: number;
    public columnSpan: number;
    public groupControlID: number;
    public groupName: string;
    public isRequired: boolean;
    public isIdentifier: boolean;
    public isForeignKey: boolean;
    public foreignFieldID: number;
    public createdDateUtc: Date;
    public lastModifiedDateUtc: Date;
    public lastModifiedBy: number;
}
export class EditFormFields extends FormField {

}
export class DeleteFormFields extends FormField {

}
export class EditFormFieldsView extends FormField {
    public readOnly: boolean;
    public hidden: boolean;
    public viewName: number;
    public row: number;
    public column: number;
    public viewID: number;
    public fieldID: number;
    public formFieldViewID: number;
}

export class FieldType {
    public Id: number;
    public Name: string;
    public Description: string;
}